lib/ProductOpener/Food.pm

```diff
@@ -1260,21 +1265,45 @@ normalize_serving_size(2.5kg)              returns 2500
 =cut
 
 sub normalize_serving_size($) {
-
 	my $serving = shift;
 
-	my $q = 0;
-	my $u;
-
-	if ($serving =~ /((\d+)(\.|,)?(\d+)?)( )?($units)\b/i) {
```

Comment:
One easy solution would be to add grams to $units, which is defined at the top of Food.pm:

my $international_units = qr/kg|g|mg|µg|oz|l|dl|cl|ml|(fl(\.?)(\s)?oz)/i;
# Chinese units: a good start is https://en.wikipedia.org/wiki/Chinese_units_of_measurement#Mass
my $chinese_units = qr/
	(?:[\N{U+6BEB}\N{U+516C}]?\N{U+514B})|  # 毫克 or 公克 or 克 or (克 kè is the Chinese word for gram)
	                                        #                      (公克 gōngkè is for "metric gram")
	(?:\N{U+516C}?\N{U+65A4})|              # 公斤 or 斤 or         (公斤 gōngjīn is a "metric kg")
	(?:[\N{U+6BEB}\N{U+516C}]?\N{U+5347})|  # 毫升 or 公升 or 升     (升 is liter)
	\N{U+5428}                              # 吨                    (ton?)
	/ix;
my $russian_units = qr/г|мг|кг|л|дл|кл|мл/i;
my $units = qr/$international_units|$chinese_units|$russian_units/i;

Reply:
Are we going to support names for other units besides grams? I feel this easier solution should be passable with just grams but maybe hard to maintain with aliases for other units too.

Reply:
Ideally, we should support all units, in all languages. eg. "50 grammes" in French. We could add the unit names in common/common.pot so that they get translated in all languages.

Reply:
Apologies for my absence over the last day, I'll work on adding the translations and then we could maybe also work on an autogeneration logic for the unit equivalence table/list.